File: Java/SpaceVIL/spacevil/src/main/java/com/spvessel/spacevil/Core/TextInputArgs.java
package com.spvessel.spacevil.Core;

import java.util.LinkedList;
import java.util.List;

import com.spvessel.spacevil.Flags.KeyMods;

/**
 * A class that describe keyboard text typing input.
 */
public final class TextInputArgs implements InterfaceInputEventArgs {
    /**
     * Character code.
     */
    public Integer character;

    /**
     * Used modifiers while typing.
     */
    public List<KeyMods> mods;

    /**
     * Clearing TextInputArgs.
     */
    public void clear() {
        character = 0;
        mods = new LinkedList<>();
        mods.add(KeyMods.NO);
    }
}

File: SpaceVIL/Core/TextInputArgs.cs
using System;
using System.Collections.Generic;

namespace SpaceVIL.Core
{
    /// <summary>
    /// A class that describe keyboard text typing input.
    /// </summary>
    public sealed class TextInputArgs : IInputEventArgs
    {
        /// <summary>
        /// Character code.
        /// </summary>
        public UInt32 Character;

        /// <summary>
        /// Used modifiers while typing.
        /// </summary>
        public KeyMods Mods;
        
        /// <summary>
        /// Clearing TextInputArgs.
        /// </summary>
        public void Clear()
        {
            Character = 0;
            Mods = 0;
        }

        public override String ToString()
        {
            return
                Character + " " +
                Mods + " ";
        }
    }
}